src/zenml/models/v2/core/component.py

```diff
@@ -124,22 +124,63 @@ def name_cant_be_a_secret_reference(cls, name: str) -> str:
         return name
 
 
-@server_owned_request_model
 class InternalComponentRequest(ComponentRequest):
     """Internal component request model."""
 
-    pass
+    user: Optional[UUID] = Field(  # type: ignore[assignment]
+        title="The id of the user that created this resource.",
+        default=None,
+    )
 
 
 # ------------------ Update Model ------------------
 
 
-@update_model
-class ComponentUpdate(ComponentRequest):
```

Comment:
This is the question to ALL changes in the update models. Why do you avoid inheriting from 2 classes, but duplicate fields instead? Are there any significant changes? Before it was working by inheriting from the Request.
If you struggle with None defaults, maybe an after validator checking for set fields and nulling out the rest would help?

Reply:
We were having a lot of issues with the previous versions of the `UpdateModel`s.

- First of all, the assumption that was made by the previous decorator was not valid for all use cases. There were some use cases before where we had to duplicate some fields again just to make them `required` for instance. 

- This is a small issue, but when you think about it, their schemas also do not match one-to-one. While request models feature fields like `user` or `workspace`, fields like this are generally not updateable and completely ignored in the backend.

- Moreover, pydantic has changed how they deal with `Optional` values. Their `Field`s do not have attributes like `required` or `allow_none` anymore. We used to manipulate these attributes to make fields optional.

- Instead, they now have a method called `is_required()` which checks if they have a `default` or a `default_factory`. So, I thought, if I manually changed them, we would end up with the same logic. Unfortunately, this was not the case. Even if you manipulate these attributes in the decorator, when you try to instantiate a model, it still says `Field required`.

- **And to my most important point**: I have no idea how this did not create any other problems before but `mypy` completely goes crazy with the old decorator. Look at this for instance:

```python
class Request(BaseModel):
    a: int = Field(title='some int field')


@update_model
class Update(Request):
    pass


def some_func(u: Update):
    return u.a + 3
```

with our previous update models, `mypy` does not see an issue here. However, if you change it to:

```python
class Request(BaseModel):
    a: int = Field(title='some int field')


class Update(BaseModel):
    a: Optional[int] = Field(title='some int field', default=None)


def some_func(u: Update):
    return u.a + 3
```

it correctly spots: `error: Unsupported operand types for + ("None" and "int")`. This led to a lot of silenced errors in our codebase and the current changes also fix them.

- Lastly, setting the fields to `None` manually in a validator would also not work. Because we use parameters like `exclude_unset` when we are updating the schemas. A field that was not provided would be updated to None even if you had nothing to do with it.

Reply:
All good, thanks for clarifications!